exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/profile/FragmentWrapper.java

```diff
@@ -49,58 +51,135 @@ public String getId() {
+    tb.appendMillis(shortRun.getEndTime() - shortRun.getStartTime(), null, null);
+    tb.appendMillis(totalDuration / complete.size(), null, null);
+    tb.appendMillis(longRun.getEndTime() - longRun.getStartTime(), null, null);
+
+    tb.appendPercent(totalProcessInMillis / (totalProcessInMillis + totalWaitInMillis), null,
+      String.format("&#8721;Busy: %,.2fs + &#8721;Wait: %,.2fs", totalProcessInMillis/1E3, totalWaitInMillis/1E3));
 
     final MinorFragmentProfile lastUpdate = Collections.max(complete, Comparators.lastUpdate);
-    tb.appendTime(lastUpdate.getLastUpdate(), null);
+    tb.appendMillis(System.currentTimeMillis()-lastUpdate.getLastUpdate(), null, new SimpleDateFormat().format(new Date(lastUpdate.getLastUpdate())));
 
     final MinorFragmentProfile lastProgress = Collections.max(complete, Comparators.lastProgress);
-    tb.appendTime(lastProgress.getLastProgress(), null);
+    tb.appendMillis(System.currentTimeMillis()-lastProgress.getLastProgress(), null, new SimpleDateFormat().format(new Date(lastProgress.getLastProgress())));
+
+    // TODO(DRILL-3494): Names (maxMem, getMaxMemoryUsed) are misleading; the value is peak memory allocated to fragment
+    final MinorFragmentProfile maxMem = Collections.max(complete, Comparators.fragmentPeakMemory);
+    tb.appendBytes(maxMem.getMaxMemoryUsed(), null, null);
+  }
+
+  public static final String[] COMPLETED_FRAGMENT_OVERVIEW_COLUMNS = {"Major Fragment", "Minor Fragments Reporting",
+    "First Start", "Last Start", "First End", "Last End", "Min Runtime", "Avg Runtime", "Max Runtime", "% Busy", "Max Peak Memory"};
+
+  public static final String[] COMPLETED_FRAGMENT_OVERVIEW_COLUMNS_TOOLTIP = {null, "# Minor Fragments Spawned",
+    null, null, null, null, "Shortest duration of a fragment", "Avg duration of a fragment", "Longest duration of a fragment", "%time Fragments were Busy", null};
+
+  //Not including Major Fragment ID and Minor Fragments Reporting
+  public static final int NUM_NULLABLE_COMPLETED_OVERVIEW_COLUMNS = COMPLETED_FRAGMENT_OVERVIEW_COLUMNS.length - 2;
+
+  public void addFinalSummary(TableBuilder tb) {
+
+    // Use only minor fragments that have complete profiles
+    // Complete iff the fragment profile has at least one operator profile, and start and end times.
+    final List<MinorFragmentProfile> complete = new ArrayList<>(
+      Collections2.filter(major.getMinorFragmentProfileList(), Filters.hasOperatorsAndTimes));
+
+    tb.appendCell(new OperatorPathBuilder().setMajor(major).build(), null, null);
+    tb.appendCell(complete.size() + " / " + major.getMinorFragmentProfileCount(), null, null);
+
+    // If there are no stats to aggregate, create an empty row
+    if (complete.size() < 1) {
+      tb.appendRepeated("", null, NUM_NULLABLE_COMPLETED_OVERVIEW_COLUMNS, null);
+      return;
+    }
+
+    final MinorFragmentProfile firstStart = Collections.min(complete, Comparators.startTime);
```

Comment:
Yes. This looks at only completed fragments (i.e. we know their start and end times).